sdk/search/azure-search/src/main/java/com/azure/search/DataSources.java

```diff
@@ -0,0 +1,264 @@
+     * @param description Optional. Description of the data source
+     * @param deletionDetectionPolicy Optional. The data deletion detection policy for the data source
+     * @return A new Azure Blob DataSource instance.
+     */
+    public static DataSource azureBlobStorage(
+        String name,
+        String storageConnectionString,
+        String containerName,
+        String pathPrefix,
+        String description,
+        DataDeletionDetectionPolicy deletionDetectionPolicy) {
+        return new DataSource()
+            .setName(name)
+            .setType(DataSourceType.AZURE_BLOB)
+            .setCredentials(new DataSourceCredentials()
+                .setConnectionString(storageConnectionString))
+            .setContainer(new DataContainer()
+                .setName(containerName)
+                .setQuery(pathPrefix))
+            .setDescription(description)
+            .setDataDeletionDetectionPolicy(deletionDetectionPolicy);
+    }
+
+    /**
+     * Creates a new DataSource to connect to an Azure Blob container.
+     *
+     * @param name The name of the data source.
+     * @param storageConnectionString The connection string for the Azure Storage account.
+     * It must follow this format: "DefaultEndpointsProtocol=https;AccountName=[your storage account];
+     * AccountKey=[your account key];" Note that HTTPS is required.
+     * @param containerName The name of the container from which to read blobs.
+     * @return A new Azure Blob DataSource instance.
+     */
+    public static DataSource azureBlobStorage(
+        String name,
+        String storageConnectionString,
+        String containerName) {
+        return DataSources.azureBlobStorage(name, storageConnectionString, containerName, null, null, null);
+    }
+
+    /**
+     * Creates a new DataSource to connect to an Azure Table.
+     *
+     * @param name The name of the data source.
+     * @param storageConnectionString The connection string for the Azure Storage account.
+     * It must follow this format: "DefaultEndpointsProtocol=https;
+     * AccountName=[your storage account];AccountKey=[your account key];"
+     * Note that HTTPS is required.
+     * @param tableName The name of the Azure table from which to read rows.
+     * @param query Optional. A query that is applied to the table when reading rows.
+     * @param description Optional. Description of the data source
+     * @param deletionDetectionPolicy Optional. The data deletion detection policy for the data source.
+     * @return A new DataSource instance.
+     * @throws IllegalArgumentException if name, tableName or storageConnectionString are null or empty.
+     */
+    public static DataSource azureTableStorage(
+        String name,
+        String storageConnectionString,
+        String tableName,
+        String query,
+        String description,
+        DataDeletionDetectionPolicy deletionDetectionPolicy) {
+        if (CoreUtils.isNullOrEmpty(name)) {
```

Comment:
Why do we have the null or empty checks for table and not for the other data sources?

Reply:
updated for sql and blob (already avail for cosom)